pkg/agent/route/route_linux.go

```diff
@@ -350,16 +335,6 @@ func writeLine(buf *bytes.Buffer, words ...string) {
 	}
 }
 
-func disableICMPSendRedirects(intfName string) error {
-	cmdStr := fmt.Sprintf("echo 0 > /proc/sys/net/ipv4/conf/%s/send_redirects", intfName)
```

Comment:
As part of the flow exporter, we make sure a couple of netfilter/*conntrack* parameters to be set.
 https://github.com/vmware-tanzu/antrea/blob/5f65bba1b771b6fd1f1b548a3458bd3647c74489/pkg/agent/flowexporter/connections/conntrack_linux.go#L151
I think the change of privileges for the antrea-agent container will affect those checks, right?

Reply:
thanks @srikartati for raising this, I didn't know we have other places to perform sysctl. Is the conntrack configuration must-have to make flow exporter work? If so, I'm thinking two approaches:
1. Keep privilege in the upstream yaml and use customized manifest for clusters that do not allow privileged pod (but flow exporter and other features that require sysctl configuration won't be available in these clusters).
2. Try performing sysctl in antrea-agent and if it fails, logging a warning but do not exit the program if an env variable like IGNORE_SYSCTL_ERR is set. This means user doesn't give antrea-agent pod privilege, assuming sysctl will be configured out-of-band.

@antoninbas @jianjuns @srikartati any idea?

Reply:
Thanks for catching this @srikartati 

* these parameters are not must-have, but they do provide additional information. The code already does not fail and exit if the kernel parameters cannot be set, so maybe `IGNORE_SYSCTL_ERR` is not required if we go with Quan's idea. Although the problem with *always* ignoring it is we do not catch issues when testing, as was the case here.
* a possibility is to use an initContainer that's privileged in order to set these kernel parameters; and run the Agent as unprivileged. The initContainer can easily be removed from the manifest when needed.
* could we consider setting the sysctls in the Pod spec as that doesn't require privileged mode (https://kubernetes.io/docs/tasks/administer-cluster/sysctl-cluster/#setting-sysctls-for-a-pod)? That relies on container runtime support to set the kernel parameters.

Reply:
@tnqn Setting up the conntrack parameters is not a must per say, but the flow visibility would be not complete especially because of the lack of flow stats. Currently, if there is no write access to sysctl parameters and the values are not desired, then we just log an error and continue. In this case, we just provide flow records without stats and timestamps.

Since having no privileges doesn't break functionality, IMO we can go for option 2 to configure them out-of-band and describe the importance of setting required sysctl parameters in the flow exporter documentation.
I would like to hear others opinion too.

Reply:
The 3rd option doesn't work as setting unsafe-sysctls need extra argument `--allowed-unsafe-sysctls` set to kubelet and it's not allowed to set net namespace related kernel parameter for a host network Pod: https://github.com/kubernetes/kubernetes/blob/2046f4212a355a3381da20c4d800283b98e9a081/pkg/kubelet/sysctl/whitelist.go#L89-L91.

I think the initContainer approach is feasible, we may need to measure how long would it add to the start time if having another initContainer. Anyway I could reduce the scope of this PR to remove send_redirects configuration only first.

Reply:
What I like about the initContainer approach is we can group all operations that require privileged mode and we ensure that the antrea-agent container itself can run without privileged mode since the default YAML itself will not include `privileged: true`. But yes, that's a separate question that can be resolved in a future PR. After merging this PR, one can already edit the manifest and remove `privileged: true`, which will cause a warning to be logged if FlowExporter is enabled, but will not terminate the agent.